### PR TITLE
Limit access to ssh node to bastion IP only

### LINKED ISSUE #257: Bastion host IP missing
**Describe the bug**
Bastion IP missing which cause security group create failure. 

```
(2/4) Setting up bastion host security group
Bastion host security group set up.

An error occurred (InvalidParameterValue) when calling the AuthorizeSecurityGroupIngress operation: CIDR block /32 is malformed
exit status 2
```

**To Reproduce**
gardenctl ssh hostname

**Expected behavior**
gardenctl ssh bastion host

**Screenshots**

**Gardenctl Version (please complete the following information):**
 - Version/Commit [latest master]

**Additional context**

PR:  https://github.com/gardener/gardenctl/pull/258

### PR DESCRIPTION
**Pull request by jfortin-sap-zz**
**What this PR does / why we need it**:
Limit access to ssh node to bastion IP only
**Which issue(s) this PR fixes**:
Fixes #257 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Limit access to ssh node to bastion IP only
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @jfortin-sap for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: pkg/cmd/ssh_aws.go [hunk 1]
```diff
@@ -63,15 +63,15 @@ func sshToAWSNode(nodeName, path, user, pathSSKeypair string, sshPublicKey []byt
 	fmt.Println("Data fetched from target shoot cluster.")
 	fmt.Println("")
 
-	fmt.Println("(2/4) Setting up bastion host security group")
+	fmt.Println("(2/4) Creating bastion host")
+	a.createBastionHostInstance()
```
Comment (tedteng) [inline pkg/cmd/ssh_aws.go:67]: inside of createBastionHostInstance,  it requre the `a.BastionSecurityGroupID` however it created at steps 3/4  
https://github.com/gardener/gardenctl/blob/c58b2124d44340f2e22fc5a4b4ae969d34c55840/pkg/cmd/ssh_aws.go#L260

so which causes the `ssh` failure.

File: pkg/cmd/ssh_aws.go [hunk 2]
```diff
@@ -181,7 +181,7 @@ func (a *AwsInstanceAttribute) createBastionHostSecurityGroup() {
 	fmt.Println("Bastion host security group set up.")
 
 	// add shh rule to ec2 instance
-	arguments = fmt.Sprintf("aws ec2 authorize-security-group-ingress --group-id %s --protocol tcp --port 22 --cidr 0.0.0.0/0", a.SecurityGroupID)
+	arguments = fmt.Sprintf("aws ec2 authorize-security-group-ingress --group-id %s --protocol tcp --port 22 --cidr %s/32", a.SecurityGroupID, a.BastionIP)
```
Comment (tedteng) [inline pkg/cmd/ssh_aws.go:184]: just realize the a.Bastion IP is Public IP address, You may need create new value for internal Bastion IP in here.

**Comment by jfortin-sap** [comment]
@tedteng Fixed

**Comment by neo-liang-sap** [comment]
@jfortin-sap could you please squash commits into one? thanks!

**Comment by tedteng** [comment]
Looks good, from the AWS console I can see the security group with correct setup in host node. 

but not sure whether APJ complex network situation, I can't ssh it. I am assuming it works from your side when you testing?
```
Downloaded id_rsa key
Check Public IP

(1/4) Fetching data from target shoot cluster
Data fetched from target shoot cluster.

(2/4) Creating bastion host
Bastion host instance i-0e216d43e971e5541 Initializing.

Bastion host instance running.
(3/4) Setting up bastion host security group
Bastion host security group set up.
Opened SSH Port on Node.

Waiting 60 seconds for Bastion SSH port open
=>
=>
=>
=>
=>
=>
SSH Port Open on Bastion TimeOut
(4/4) Cleanup
Cleaning up bastion host configurations...

Starting cleanup

  (1/3) Cleaning up bastion host instance
TERMINATINGINSTANCES	i-0e216d43e971e5541
CURRENTSTATE	32	shutting-down
PREVIOUSSTATE	16	running

  (2/3) Close SSH Port on Node.
  Closed SSH Port on Node.

  (3/3) Clean up bastion host security group


Bastion host configurations successfully cleaned up.
```

**Comment by jfortin-sap** [comment]
@tedteng I get the same issue with or without the fix.

**Comment by tedteng** [comment]
I will check it later

**Comment by tedteng** [comment]
It seems something happed during the process
as you can see the result sg id is `sg-0942a3d62d289be2d` (I am also change 0.0.0.0/0 just for testing)
image

but in `bsg` instance it is another one binding and not 22 rule, which means port 22 never open  
image

the real sg seems not biding with the `bsg` instance.
image


if you don't mind I can adjust it later after my PR https://github.com/gardener/gardenctl/pull/263 merge.

**Comment by jfortin-sap** [comment]
@tedteng Ok, let me know when ready to merge

**Comment by tedteng** [comment]
@jfortin-sap  I think I figure it out, The ec2 instance will create default security group when instance created. that why the ssh-access security group can't attach the bastion instance.

**Comment by neo-liang-sap** [comment]
@jfortin-sap @tedteng could we close this issue as https://github.com/gardener/gardenctl/commit/5dd0947b3d52ec0f4d17252c7d96bf9efdf2e3f9 is already in master branch?

**Pull request closed without merging**